android/src/main/java/com/mrousavy/camera/CameraView.kt

```diff
@@ -410,9 +411,15 @@ class CameraView(context: Context, private val frameProcessorThread: ExecutorSer
         // User has selected a custom format={}. Use that
         val format = DeviceFormat(format!!)
         Log.i(TAG, "Using custom format - photo: ${format.photoSize}, video: ${format.videoSize} @ $fps FPS")
-        previewBuilder.setTargetResolution(format.videoSize)
+        val videoSize:Size
+        if (context.resources.configuration.orientation == Configuration.ORIENTATION_PORTRAIT) {
```

Comment:
Hey, thanks again for the PR.

Unfortunately I am still not quite sure if I understand this correctly, doesn't this only check the "locked" Orientation in AndroidManifest? What if the user is holding his phone in portrait, then going to landscape? And then going back to portrait? Is it still the correct resolution?

Reply:
If I read the docs correctly, `context.resources.configuration.orientation` returns the device's current orientation.

Reply:
Okay, but can we use `outputOrientation` instead of `context.resources.configuration.orientation` here? I already detect orientation in this class

Reply:
`outputOrientation` is referencing `context.displayRotation`. That tells us how many degrees the device is rotated from it's natural position, but not if we're looking at the device in a portrait or landscape orientation.

`context.resources.configuration.orientation` differs as it tells us whether the device is in portrait or landscape.

Maybe an issue here too. Portrait would only equal Surface.ROTATION_0 for naturally-portrait devices:

```kotlin
  private val outputRotation: Int
    get() {
      if (orientation != null) {
        // user is overriding output orientation
        return when (orientation!!) {
          "portrait" -> Surface.ROTATION_0
          "landscapeRight" -> Surface.ROTATION_90
          "portraitUpsideDown" -> Surface.ROTATION_180
          "landscapeLeft" -> Surface.ROTATION_270
 ```

Reply:
I cannot find `outputOrientation`. There is `orientation` which is a `string` and `outputRotation` which is an integer.

Reply:
Sorry, typo on my end. I fixed up my earlier post. I meant `outputRotation`

Reply:
Okay guys thanks for the insights. I will test this some time this week and then merge the PR if everything works in every use-case :)

Would be helpful if anyone could help me with testing, with various devices and various orientations (e.g. start with landscape or portrait, then rotate)

Reply:
@mrousavy What are your views on this PR, is it ready to get merged or ...?